src/meta/upgradeData/MetaDataUpgrade.cpp

```diff
@@ -251,6 +251,29 @@ Status MetaDataUpgrade::convertToNewColumns(const std::vector<oldmeta::cpp2::Col
     return Status::OK();
 }
 
+Status
+MetaDataUpgrade::convertToNewIndexColumns(const std::vector<oldmeta::cpp2::ColumnDef> &oldCols,
+                                          std::vector<cpp2::ColumnDef> &newCols) {
+    for (auto &colDef : oldCols) {
+        cpp2::ColumnDef columnDef;
+        columnDef.set_name(colDef.get_name());
+        if (colDef.get_type().get_type() == oldmeta::cpp2::SupportedType::STRING) {
+            cpp2::ColumnTypeDef type;
+            type.set_type(cpp2::PropertyType::FIXED_STRING);
+            type.set_type_length(256);
+            columnDef.set_type(std::move(type));
+        } else {
+            columnDef.type.set_type(static_cast<cpp2::PropertyType>(colDef.get_type().get_type()));
+        }
+        DCHECK(!colDef.__isset.default_value);
+        if (FLAGS_null_type) {
+            columnDef.set_nullable(true);
+        }
```

Comment:
Do we really need this nullable for ColumnDef? How is this parameter set for tag|edge schema ?

Reply:
I don't sure that is the index require the null flag so I don't modify it. If it's useless I could remove it.

Reply:
> I don't sure that is the index require the null flag so I don't modify it. If it's useless I could remove it.

got, please wait a moment. let me think about it.

Reply:
> > I don't sure that is the index require the null flag so I don't modify it. If it's useless I could remove it.
> 
> got, please wait a moment. let me think about it.

I think, because v1 doesn't support nullable parameter, when  the meta upgraded from v1 to v2,  we don't need to set nullable param to true(default value `false` is enough), so I think the ColumnDef of IndexItem also don't need set it. WDYT?

Reply:
In fact, 1.0 has no null but 2.0 has it.So Laura add FLAGS_null_type to determine true or false for the null option in 2.0 by user. And for index I reserve this logical if the index require the null option or I could remove this logical.

Reply:
IMO, both add FLAGS_null_type in tag/edge and index schema, or neither of them add the flag.

Reply:
> In fact, 1.0 has no null but 2.0 has it.So Laura add FLAGS_null_type to determine true or false for the null option in 2.0 by user. And for index I reserve this logical if the index require the null option or I could remove this logical if index process don't use it.

Got, I've learned. keep it.

Reply:
Yes as @critical27 said, but if the null is useless in index process we could don't set it.